### PR TITLE
Improve benchmark compatibility with v9

### PR DESCRIPTION
**Pull request by alantreadway**
- **Fix benchmark v9.x.y backwards compatibility.**
- **Improve historic version benchmark flow.**

### CODE REVIEW

File: packages/ag-charts-community/benchmarks/benchmark.ts [hunk 1]
```diff
@@ -36,7 +46,8 @@ export class BenchmarkContext<T extends AgChartOptions = AgChartOptions> {
             await this.chart?.update(this.options);
             return;
         }
-        await (AgCharts as any).update(this.chart, this.options);
+        (AgCharts as any).update(this.chart, this.options);
+        await this.waitForUpdate();
```
Comment (jacobp100) [inline packages/ag-charts-community/benchmarks/benchmark.ts:50]: Has our API changed?

Comment (jacobp100) [reply packages/ag-charts-community/benchmarks/benchmark.ts:50]: Or is this for backwards compatibility?

**Review by jacobp100 — APPROVED** [review]

File: packages/ag-charts-community/benchmarks/benchmark.ts (hunk 1 above)
Comment (alantreadway) [reply packages/ag-charts-community/benchmarks/benchmark.ts:50]: Yes, so in 10.0.0 we added the promise-based + chart-ref centric APIs, but prior to that we didn't have either of these aspects, it appears (or the promise only waited for option application, not rendering) - for v9 updates to be timed correctly, we need to explicitly call `waitForUpdate()` to correctly wait for the chart to re-render.

Comment (alantreadway) [reply packages/ag-charts-community/benchmarks/benchmark.ts:50]: (this change is in the <10.0.0 path)

**Pull request merged**